consensus/core/src/dag_state.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    sync::Arc,
+};
+
+use crate::{
+    block::{BlockAPI as _, BlockRef, VerifiedBlock},
+    context::Context,
+    storage::Store,
+};
+
+/// DagState provides the API to write and read accepted blocks from the DAG.
+/// The underlying blocks can be cached in memory or stored on disk.
+#[allow(unused)]
+pub(crate) struct DagState {
```

Comment:
How do we intend to share the `DagState` ? Is it supposed to be accessed by a single thread (hence `Core`) only?

Reply:
Got it . I guess there could be a `DagStateInner` which could be guarded behind the lock and have the `DagState` be shared across components? I am sure you have already in your mind the approach , just wanted to see how ready the struct is to start injecting/use across components (even if API is not finalised yet)

Reply:
Yes both Mysticeti and Narwhalceti used the mutex-guarded inner value approach. But the disadvantage is that either DagState need to take in higher level logic and expose them as APIs, or DagState usages can have a lot of lock & unlock cost underneath. Maybe the cost is not that high, but I want to see if explicit lock & unlock can work. Let me know if it turns out to be a pain and I can refactor.